include/IECoreScene/PrimitiveVariable.h

```diff
@@ -102,11 +102,75 @@ struct IECORESCENE_API PrimitiveVariable
 	/// array of unique UVs in data and map them to FaceVarying using
 	/// the indices.
 	IECore::IntVectorDataPtr indices;
+
+	template<typename T>
+	class IndexedRange;
+
+};
+
+/// Utility class for iterating the `data` field from
+/// a PrimitiveVariable, using the `indices` field
+/// appropriately if it exists.
+///
+/// > Note : This intentionally only provides `const`
+/// > access because it does not make sense to modify
+/// > the `data` elements via the `indices`, since
+/// > each element will be visited an unknown number of
+/// > times.
+template<typename T>
+class PrimitiveVariable::IndexedRange
+{
+
+	public :
+
+		/// Throws if the PrimitiveVariable doesn't contain
+		/// `TypedData<vector<T>>`.
+		IndexedRange( const PrimitiveVariable &variable );
```

Comment:
Given we're passing in a struct of ref counted pointers - I think we need a few comments here that an IndexedRange's lifetime needs to be bound by the referenced PrimitiveVariable's lifetime.

Reply:
Renamed to IndexedView and added the suggested documentation in e8f37c5.